### PR TITLE
Memleak checks, and fixes.

### PR DESCRIPTION
**Pull request by rustyrussell**
Probably slows things down a fair amount, but we can make it turned off, even if DEVELOPER, using an environment var.

### CODE REVIEW

File: common/htlc_wire.c [hunk 1]
```diff
@@ -147,6 +148,9 @@ void fromwire_bitcoin_tx(const u8 **cursor, size_t *max, struct bitcoin_tx *tx)
 	 * for the sake of simple structures, we don't write the
 	 * generator that way. */
 	struct bitcoin_tx *tx2 = pull_bitcoin_tx(tx, cursor, max);
-	if (tx2)
+	if (tx2) {
 		*tx = *tx2;
+		/* This hangs around with tx until freed */
+		notleak(tx2);
```
Comment (cdecker) [inline common/htlc_wire.c]: Any reason we aren't just `tal_free`ing this one?

**Comment by cdecker** [comment]
Added a commit to fix `DEVELOPER=0` builds. No idea where the second failure came from:

```------------------------------- Valgrind errors --------------------------------
Valgrind error file: valgrind-errors.11145
==11145== Jump to the invalid address stated on the next line
==11145==    at 0x0: ???
==11145==    by 0x482296: backtrace_full (backtrace.c:127)
==11145==    by 0x40FC95: log_crash (log.c:436)
==11145==    by 0x535A4AF: ??? (in /lib/x86_64-linux-gnu/libc-2.23.so)
==11145==    by 0x535A427: raise (raise.c:54)
==11145==    by 0x535C029: abort (abort.c:89)
==11145==    by 0x4100CF: fatal (log.c:525)
==11145==    by 0x4225E8: log_status_fail (subd.c:404)
==11145==    by 0x4227D8: sd_msg_read (subd.c:454)
==11145==    by 0x457166: next_plan (io.c:59)
==11145==    by 0x457C3D: do_plan (io.c:387)
==11145==    by 0x457C7B: io_ready (io.c:397)
==11145==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==11145== 
==11145== 
==11145== Process terminating with default action of signal 11 (SIGSEGV)
==11145==  Bad permissions for mapped region at address 0x0
==11145==    at 0x0: ???
==11145==    by 0x482296: backtrace_full (backtrace.c:127)
==11145==    by 0x40FC95: log_crash (log.c:436)
==11145==    by 0x535A4AF: ??? (in /lib/x86_64-linux-gnu/libc-2.23.so)
--------------------------------------------------------------------------------
----------------------------- Captured stderr call -----------------------------
Subdaemon lightning_onchaind hit error
Log dumped in crash.log```

Full logs [here](https://api.travis-ci.org/v3/job/316859970/log.txt)

File: common/htlc_wire.c (hunk 1 above)
Comment (rustyrussell) [reply common/htlc_wire.c]: Yes, because it has children we need.  We could steal all the children onto tx, then free, but this seemed th e lesser evil.

**Comment by rustyrussell** [comment]
This failure is deliberate: the node gets upset because we spend its revoked commitment tx.  I'm trying to figure out why it's trying to backtrace with valgrind, though: we suppress that for exactly this reason.

**Comment by rustyrussell** [comment]
Rebased to fix trivial conflict.

**Comment by rustyrussell** [comment]
Ping @cdecker I think we're good now?

**Pull request merged**